### PR TITLE
Autowire models.

### PR DESCRIPTION
**Pull request by biozshock**
| Q                                      | A
| -------------------------------------- | ---
| New feature/enhancement? (use the a.x branch)      | [ x ]
| Deprecations?                          | [ x ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ x ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 

#### Description:
Provides autowired models and allows to use `ModelFactory->getModel(ModelClass::class)`.
As a side feature adds ability to mark a class as autowired in old `config.php` config file.

#### Steps to test this PR:
This is another broad change of the system, so please test several pages. 

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Check the website is working.

### CODE REVIEW

File: app/bundles/ApiBundle/Config/config.php [hunk 1]
```diff
@@ -92,10 +92,8 @@
         ],
         'models' => [
             'mautic.api.model.client' => [
-                'class'     => 'Mautic\ApiBundle\Model\ClientModel',
-                'arguments' => [
-                    'request_stack',
-                ],
+                'class'    => 'Mautic\ApiBundle\Model\ClientModel',
+                'autowire' => true,
             ],
```
Comment (escopecz) [inline app/bundles/ApiBundle/Config/config.php]: Can this and other models in this PR be moved to the services.php file instead? Something like https://github.com/mautic/mautic/blob/5.x/app/bundles/AssetBundle/Config/services.php#L23

Comment (biozshock) [reply app/bundles/ApiBundle/Config/config.php]: This is sooo tedious ;)

Comment (escopecz) [reply app/bundles/ApiBundle/Config/config.php]: I know. This all refactoring is :) I was just curious if it does the same thing. I don't like that we are adding some new logic in processing services in config.php files. I'd like to get rid of it in M6.

**Review by escopecz — CHANGES_REQUESTED** [review]
Please check the failing PHPSTAN issues. 

When I checkout this PR and clear cache, I get:
```
500 Internal Server Error - Call to a member function get() on null 
/html/app/bundles/CoreBundle/Model/NotificationModel.php:200 at
/html/app/bundles/CoreBundle/Controller/DefaultController.php:74 at Mautic\CoreBundle\Model\NotificationModel ->
```
when accessing the config form `s/config/edit`

File: app/bundles/ApiBundle/Config/config.php (hunk 1 above)
Comment (biozshock) [reply app/bundles/ApiBundle/Config/config.php]: Done.

Comment (escopecz) [reply app/bundles/ApiBundle/Config/config.php]: Wow, it looks amazing! Can you please resolve also the conflicts? I'll test it right after that and get it merged. This is not a BC break so it does not block the M5-alpha release.

**Review by escopecz — APPROVED** [review]
I did not notice any issues with the code. I can browse the administration without any errors. 

I'm not a fan of these long constructor dependency lists, but it's always been there. Just hidden. Now it's explicit and we can see that the models do way more then they should and we can refactor that.

Thank you! 👍

**Review by mabumusa1 — APPROVED** [review]
Changes looks good and the system works as expected

**Pull request merged**